Add v1.0.6 release notes (#5085)
spiceai/spiceai: docs/release_notes/v1.0.6.md

# Spice v1.0.6 (Mar 17, 2025)

> Spice v1.0.6 improves stability for DuckDB acceleration, Iceberg Data/Catalog connector improvements when using AWS Glue, and fixes an issue with the `on_registration: source` federation fallback when using DuckDB.

Author: ```suggestion
Spice v1.0.6 improves stability for DuckDB acceleration, Iceberg Data/Catalog connector improvements when using AWS Glue, and fixes an issue with the `on_registration: source` federation fallback when using DuckDB. In addition, redundant data refreshes on startup are avoided for accelerations with persistent data.
```

## Highlights in v1.0.6

- **Iceberg Data/Catalog Connector Improvements**: Improves Iceberg data & catalog connector reliability, including bug fixes for AWS Glue API rate-limiting and compatibility, REST API pagination support, explicit AWS credential handling, and support for AWS STS role assumption.

- **Fixes On-Registration Fallback when using DuckDB**: Previously, when using DuckDB as a data accelerator and the `on_registration: source` configuration, queries made during the initial data refresh did not properly fallback to the federated source. This is now fixed.

- **DuckDB downgraded for Stability**: DuckDB has been downgraded to v1.1.3 due to a regression in memory handling tracked by [duckdb/duckdb issue #16640](https://github.com/duckdb/duckdb/issues/16640). Once resolved and validated, Spice will re-upgrade to v1.2.x.

- **Expanded Integration Tests**: Additional integration tests covering federated accelerator behavior and graceful shutdown processes have been added.

> 

Author: ```suggestion
**Optimized Data Refresh for Persistent Accelerations**: Changed behavior in v1.0.6. When using persistent (file-mode) acceleration without a defined refresh interval, Spice performs a full refresh at startup only if no previously accelerated data is available. This ensures efficient startup behavior by avoiding unnecessary refreshes. This logic applies only to full refreshes when no refresh interval is specified.

To maintain the previous behavior and always refresh on every startup, set:

```yaml
 acceleration:
  refresh_on_startup: always
``` 

```

## Highlights in v1.0.6

- **Iceberg Data/Catalog Connector Improvements**: Improves Iceberg data & catalog connector reliability, including bug fixes for AWS Glue API rate-limiting and compatibility, REST API pagination support, explicit AWS credential handling, and support for AWS STS role assumption.

- **Fixes On-Registration Fallback when using DuckDB**: Previously, when using DuckDB as a data accelerator and the `on_registration: source` configuration, queries made during the initial data refresh did not properly fallback to the federated source. This is now fixed.

- **DuckDB downgraded for Stability**: DuckDB has been downgraded to v1.1.3 due to a regression in memory handling tracked by [duckdb/duckdb issue #16640](https://github.com/duckdb/duckdb/issues/16640). Once resolved and validated, Spice will re-upgrade to v1.2.x.

- **Expanded Integration Tests**: Additional integration tests covering federated accelerator behavior and graceful shutdown processes have been added.

> **Optimized Data Refresh for Persistent Accelerations**: Changed behavior in v1.0.6. When using persistent (file-mode) acceleration without a defined refresh interval, Spice performs a full refresh at startup only if no previously accelerated data is available. This ensures efficient startup behavior by avoiding unnecessary refreshes. This logic applies only to full refreshes when no refresh interval is specified.

Author: ```suggestion

**Optimized Data Refresh for Persistent Accelerations**: Changed behavior in v1.0.6. When using persistent (file-mode) acceleration without a defined refresh interval, Spice performs a full refresh at startup only if no previously accelerated data is available. This ensures efficient startup behavior by avoiding unnecessary refreshes. This logic applies only to full refreshes when no refresh interval is specified.
```

- **DuckDB downgraded for Stability**: DuckDB has been downgraded to v1.1.3 due to a regression in memory handling tracked by [duckdb/duckdb issue #16640](https://github.com/duckdb/duckdb/issues/16640). Once resolved and validated, Spice will re-upgrade to v1.2.x.

- **Expanded Integration Tests**: Additional integration tests covering federated accelerator behavior and graceful shutdown processes have been added.

**Optimized Data Refresh for Persistent Accelerations**: Changed behavior in v1.0.6. When using persistent (file-mode) acceleration without a defined refresh interval, Spice performs a full refresh at startup only if no previously accelerated data is available. This ensures efficient startup behavior by avoiding unnecessary refreshes. This logic applies only to full refreshes when no refresh interval is specified.

To maintain the previous behavior and always refresh on every startup, set:

```yaml
 acceleration:

>   refresh_on_startup: always

Author: ```suggestion
  refresh_on_startup: always
```
  
```

>   refresh_on_startup: always

Author: ```suggestion
  refresh_on_startup: always
```

```

# Spice v1.0.6 (Mar 17, 2025)

> Spice v1.0.6 improves stability for DuckDB acceleration, Iceberg Data/Catalog connector improvements when using AWS Glue, and fixes an issue with the `on_registration: source` federation fallback when using DuckDB. In addition, redundant data refreshes on startup are avoided for accelerations with persistent data.

Reviewer: ```suggestion
Spice v1.0.6 improves stability for DuckDB acceleration, Iceberg Data/Catalog connector improvements when using AWS Glue, and fixes an issue with the `ready_state: on_registration` federation fallback when using DuckDB. In addition, redundant data refreshes on startup are avoided for accelerations with persistent data.
```

## Highlights in v1.0.6

- **Iceberg Data/Catalog Connector Improvements**: Improves Iceberg data & catalog connector reliability, including bug fixes for AWS Glue API rate-limiting and compatibility, REST API pagination support, explicit AWS credential handling, and support for AWS STS role assumption.

> - **Fixes On-Registration Fallback when using DuckDB**: Previously, when using DuckDB as a data accelerator and the `on_registration: source` configuration, queries made during the initial data refresh did not properly fallback to the federated source. This is now fixed.

Reviewer: ```suggestion
- **Fixes On-Registration Fallback when using DuckDB**: Previously, when using DuckDB as a data accelerator and the `ready_state: on_registration` configuration, queries made during the initial data refresh did not properly fallback to the federated source. This is now fixed.
```
